kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kms/aws/AwsKmsLocal.java

```diff
@@ -0,0 +1,218 @@
+                Optional.of(Path.of(TestUtils.getResourcesURI("helm_localstack_overrides.yaml"))),
+                Optional.empty());
+
+        this.podName = KafkaUtils.getPodNameByLabel(deploymentNamespace, "app.kubernetes.io/name", LOCALSTACK_SERVICE_NAME, Duration.ofSeconds(30));
+        DeploymentUtils.waitForDeploymentRunning(deploymentNamespace, podName, Duration.ofMinutes(1));
+
+        if (!isCorrectVersionInstalled()) {
+            throw new KubeClusterException("AWS version installed " + getLocalStackVersionInstalled() + " does not match with the expected: '"
+                    + AwsKmsTestKmsFacade.LOCALSTACK_IMAGE.getVersionPart() + "'");
+        }
+    }
+
+    private boolean isCorrectVersionInstalled() {
+        String installedVersion = getLocalStackVersionInstalled();
+        String expectedVersion = AwsKmsTestKmsFacade.LOCALSTACK_IMAGE.getVersionPart();
+
+        return compareVersions(installedVersion, expectedVersion) == 0;
+    }
+
+    private int compareVersions(String currentVersion, String expectedVersion) {
+        Objects.requireNonNull(expectedVersion);
+
+        String[] currentParts = currentVersion.split("\\.");
+        String[] expectedParts = expectedVersion.split("\\.");
+
+        for (int i = 0; i < expectedParts.length; i++) {
+            int currentPart = i < currentParts.length ? Integer.parseInt(currentParts[i]) : 0;
+            int expectedPart = Integer.parseInt(expectedParts[i]);
+            int comparison = Integer.compare(currentPart, expectedPart);
+            if (comparison != 0) {
+                return comparison;
+            }
+        }
+        return 0;
+    }
+
+    @Override
+    public void delete() {
+        LOGGER.info("Deleting Aws in {} namespace", deploymentNamespace);
+        NamespaceUtils.deleteNamespaceWithWait(deploymentNamespace);
+    }
+
+    @Override
+    public URI getAwsUrl() {
+        var nodeIP = kubeClient(deploymentNamespace).getClient().nodes().list().getItems().get(0).getStatus().getAddresses().get(0).getAddress();
+        var spec = kubeClient().getService(deploymentNamespace, LOCALSTACK_SERVICE_NAME).getSpec();
+        int port = spec.getPorts().stream().map(ServicePort::getNodePort).findFirst()
+                .orElseThrow(() -> new KubeClusterException("Unable to get the service port of Aws"));
+        String url = nodeIP + ":" + port;
+        LOGGER.debug("AWS URL: {}", url);
+        return URI.create("http://" + url);
+    }
+
+    @Override
+    public String getRegion() {
+        try (var output = new ByteArrayOutputStream();
+                var error = new ByteArrayOutputStream();
+                var exec = kubeClient().getClient().pods()
+                        .inNamespace(deploymentNamespace)
+                        .withName(podName)
+                        .writingOutput(output)
+                        .writingError(error)
+                        .exec("sh", "-c", AWS_LOCAL_CMD + " configure get region")) {
```

Comment:
why are you using the  shell rather than exec the arguments directly.

```
.exec(AWS_LOCAL_CMD, "configure", "get", "region")
```

it should be equal and saves a fork/exec

Reply:
It didn't work for Vault and honestly I didn't test it here for AWS. Will test it

Reply:
refactored to use `execInPod` instead